ci/unit_tests/functions_deploy/main_test.py

```diff
@@ -0,0 +1,197 @@
+"""
+Copyright 2019 IBM Corporation
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json, os, requests, shutil, tempfile, unittest, urllib, uuid, zipfile
+
+import functions_deploy
+from ...test_utils import BaseTestCaseCapture
+
+class TestMain(BaseTestCaseCapture):
+
+    dataBasePath = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'main_data')
+    namespace = "Prague Cognitive Services_IoT-Prague"
+    urlNamespace = urllib.quote(namespace)
+    packageBase = "Package-for-WAW-CI-"
+
+    def callfunc(self, *args, **kwargs):
+        functions_deploy.main(*args, **kwargs)
+
+    def _getFunctionsInPackage(self, package):
+        functionListUrl = 'https://us-south.functions.cloud.ibm.com/api/v1/namespaces/' + self.urlNamespace + \
+            '/actions/?limit=0&skip=0'
+
+        functionListResp = requests.get(functionListUrl, auth=(os.environ['CLOUD_FUNCTIONS_USERNAME'],
+                                                               os.environ['CLOUD_FUNCTIONS_PASSWORD']),
+                                        headers={'accept': 'application/json'})
+
+        assert functionListResp.status_code == 200
+
+        functionListJson = functionListResp.json()
+        functionNames = []
+
+        for function in functionListJson:
+            if (self.namespace + '/' + package) in function['namespace']:
+                functionNames += [function['name']]
+
+        return functionNames
+
+
+    def setup_method(self):
+        BaseTestCaseCapture.checkEnvironmentVariables(['CLOUD_FUNCTIONS_USERNAME', 'CLOUD_FUNCTIONS_PASSWORD'])
+        self.package = self.packageBase + str(uuid.uuid4())
+        self.packageCreated = False # test should set that to true if it created package for cloud functions
+        self.dirsToDelete = [] # directories that will be deleted in tearDown method
```

Comment:
Consider putting by-product into `output` folder to be stored in artifactory.

Reply:
changed